database/README.md

```diff
@@ -1,14 +1,414 @@
-This document explains the database that the orderbook api uses.
+Here we document the current state of the database. The history of these changes lives in the `sql` folder which contains all migrations. This document shows the schema and the purpose of the tables.
 
-We need to store orders persistently even if the orderbook service goes down. For this we use a PostgreSQL database. The schema can be found in `schema.sql`.
+Code that directly interfaces with the database through SQL queries lives in the `database`. This crate is often wrapped into higher level components by consumers.
 
-In addition to the orders, we also store trade events. This allows us to keep track of filled amounts.
+With a live database information for all tables can be retrieved with the `\d` command and information for a specific table with `\d MyTable`.
 
-To prevent users from spamming the database we should when receiving an order check that
-* the erc20 spending is approved
-* valid_to is not in the past and not more than <1 year> in the future
-* the order owner does not already have more than <1000> valid orders
+Some tables only store data emitted via smart contract events. Because we only have a single deployment of the [`GPv2Settlement`](https://github.com/cowprotocol/contracts/blob/main/src/contracts/GPv2Settlement.sol) settlement contract shared across staging and production environments events related to staging **and** production orders and settlements will be present in **both** the staging **and** production databases.  
+[CoWSwapEthFlow](https://github.com/cowprotocol/ethflowcontract/blob/main/src/CoWSwapEthFlow.sol) we actually deployed twice so events related to the staging environment should only show up in the staging DB and likewise for production.  
+It's also important to note that we only index events from blocks that we are certain will not get reorged. That means specifically that events will be indexed with a block delay of at least 64.  
 
-This ensures that the amount of orders we have to keep track of per user is limited and that there is a real cost the user incurs before being able to submit orders because they need to have sent an ethereum transaction.
+### auction\_participants
+
+This table is used for [CIP-20](https://snapshot.org/#/cow.eth/proposal/0x2d3f9bd1ea72dca84b03e97dda3efc1f4a42a772c54bd2037e8b62e7d09a491f). It stores which solvers (identified by ethereum address) participated in which auctions (identified by auction id). CIP-20 specifies that "solver teams which consistently provide solutions" get rewarded.
+
+   Column     |  Type  | Nullable | Details
+--------------|--------|----------|--------
+ auction\_id  | bigint | not null | id of the auction
+ participant  | bytea  | not null | solver that submitted a **valid** solution for the auction
+
+Indexes:  
+- PRIMARY KEY: btree(`auction_id`, `participant`)
```

Comment:
Documenting the indexes for each table seems like a huge maintenance burden. Especially considering that these indexes are mostly an optimization. I'd leave them out TBH.

Reply:
I'd like to keep the indexes. They are important design choices.